lib/woothee/os.rb

```diff
@@ -89,7 +89,7 @@ def self.challenge_linux(ua, result)
     os_version = nil
     if ua.index('Android')
       data = Woothee::DataSet.get('Android')
-      if ua =~ /Android[- ](\d+\.\d+(?:\.\d+)?)/
+      if ua =~ /Android[- ](\d+\.*\d*(?:\.\d+)?)/
```

Comment:
I think it should rather be:
```suggestion
      if ua =~ /Android[- ](\d+(\.\d+(?:\.\d+)?)?)/
```

Reply:
This code is better because of less captures.
```suggestion
      if ua =~ /Android[- ](\d+(?:\.\d+(?:\.\d+)?)?)/
```